test/release/examples/primers/loops.chpl

```diff
@@ -0,0 +1,950 @@
+  ability to implement the loop using those instructions.
+
+  Like the for-loops above, Chapel's foreach-loops support zippered
+  iteration.  For example, the following loop performs a zippered
+  iteration over the array `A` and an unbounded range `1..`.
+
+*/
+
+  foreach (a, i) in zip(A, 1..) do
+    a += (i / 100.0);
+
+   writeln("After our first zippered foreach-loop, A is: ", A);
+
+/*
+
+  Note that if a Chapel program only used foreach-loops to express its
+  parallelism, it would never make use of the multiple processor cores
+  of a modern processor nor the distinct compute nodes of a cluster or
+  HPC system.  This is because foreach-loops don't ever introduce new
+  Chapel tasks, and tasks are the only way to run in parallel at
+  system scales beyond a single processor.  As a result, to leverage
+  the full power of most parallel platforms, we need to look to
+  Chapel's other parallel loop forms.
+
+
+  forall-loops
+  ------------
+
+  Forall-loops are similar to foreach-loops, except that they have the
+  potential to be implemented using multiple Chapel tasks.  This
+  permits them to use multiple cores and/or compute nodes to execute
+  the loop's iterations.
+
+  Just as Chapel's for-loops invoke a serial iterator, its
+  forall-loops invoke a parallel iterator.  Where serial iterators may
+  only yield values sequentially, a parallel iterator's yield
+  statements may occur within parallel loops and constructs, resulting
+  in parallel execution.  When using a forall-loop with zippered
+  iterands, the first iterand in the zippering controls the loop's
+  parallelism.  For details about writing such parallel iterators, see
+  the :ref:`Parallel Iterators Primer <primers-parIters>`.
+
+  A parallel iterator can create as many tasks as it wants, and can
+  specify where they should run.  By convention, most will create as
+  many tasks as are appropriate for the hardware parallelism that the
+  loop iterand targets.  For example, the default parallel iterator
+  for a range, local domain, or local array typically implements its
+  iterations using a number of tasks equal to the number of local
+  processor cores that are available, since those data structures are
+  stored on a single locale.  In contrast, the default parallel
+  iterator for a distributed domain or array will typically implement
+  the iterations using all of the available processor cores on all of
+  the locales that own a subset of the domain's indices or array's
+  elements.
+
+  It is guaranteed that all tasks created by the parallel iterator to
+  run the loop's iterations will complete before the original task
+  encountering the forall-loop proceeds.  Logically, you can think of
+  there as being a logical *join* operation on all tasks that are
+  helping to implement the forall-loop.
+
+  Looking at some simple examples, when run on a k-core processor, for
+  large enough values of ``n``, each of the following loops will
```

Comment:
Recommending putting this "for large enough values of ``n``" in a parenthetical at the end of the sentence.  I had to read the sentence three times to parse the main point.